hpx/runtime/threads/coroutines/detail/context_linux_x86.hpp

```diff
@@ -82,16 +82,6 @@ namespace hpx { namespace threads { namespace coroutines
 
     namespace detail { namespace lx
     {
-        template <typename TO, typename FROM>
-        TO nasty_cast(FROM f)
-        {
-            union {
-                FROM f; TO t;
-            } u;
-            u.f = f;
-            return u.t;
-        }
```

Comment:
@sithhell, @K-ballo, @hkaiser I'm trying to sneak this one in at the same time. AFAICT `reinterpret_cast` will do the right thing where `nasty_cast` was used, but am I missing some weird corner of the standards?